test/test_torch.py

```diff
@@ -7290,6 +7290,55 @@ def run_test(p):
         with self.assertRaisesRegex(RuntimeError, "chain_matmul: Expected one or more matrices"):
             torch.chain_matmul()
 
+    # no CUDA LU yet, used for torch.det())
+    @onlyCPU
+    @skipCPUIfNoLapack
+    @dtypes(torch.complex64, torch.complex128)
+    def test_det_complex_only(self, device, dtype):
+        scalar_dtype = {
+            torch.complex64: torch.float32,
+            torch.complex128: torch.float64
+        }
+
+        def run_test(*sizes):
+            x = torch.rand(*sizes, device=device, dtype=dtype)
+            x = x.view(-1, sizes[-2], sizes[-1])
+
+            if TEST_NUMPY:
+                x_detached = x.detach()
+                for i in range(x.size(0)):
+                    det_numpy = np.linalg.det(x_detached.select(0, i).cpu().numpy())
+                    det_pytorch = x.select(0, i).det()
+                    self.assertEqual(
+                        det_pytorch, det_pytorch.new_tensor(det_numpy),
```

Comment:
`torch.from_numpy(det_numpy).to(det_pytorch.device)`, we are trying to not use `new_tensor` as it is an old style API. Alternatively, you can use `compare_with_numpy` here.